### PR TITLE
feat: add webi-essentials - ensures curl, zip, etc exist & do shell integrations

### PR DESCRIPTION
**Pull request by coolaj86**
Installs curl, git, tar, wget, xz, and zip on Mac, Linux, Alpine (Busybox), and Windows!

- macOS
  - [x] detects and waits for Rosetta
  - [x] detects and waits for Command Line Developer Tools 🦾
  - [x] installs xz via Webi
- Linux
  - [x] only prompts for sudo if needed
  - [x] skips sudo prompt if not a TTY (i.e. over ssh or part of deploy script)
  - [x] properly detects and uses curl, wget, and busybox-wget
- Windows
  - [x] checks for curl.exe (wget), tar.exe (no zip)
  - [x] installs git & xz via Webi

For a later commit

- Shell integrations will be part of
  - #742 \
  (but we want to tell about them now so it's not a surprise)
  
# Test Matrix

TTY detection is now 😘👨‍🍳 perfect!

- [x] macOS Ventura & Sonoma (via UTM)
- [x] Ubuntu (via Proxmox)
- [x] Ubuntu over ssh (no TTY)
- [x] Alpine (Busybox) (via Proxmox)
- [x] Alpine (Busybox) over ssh (no TTY)
- [x] Windows 11 via UTM

### CODE REVIEW

File: webi-essentials/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+---
+
+### Cheat Sheet
+
+> Even Webi needs a place to start.
+
+This installs the tools that are commonly needed to download and unpack various
+packages, using the native package manager.
+
+**This requires root or `sudo`**
```
Comment (HacDan) [inline webi-essentials/README.md:15]: May be better to add something about Administrator access, being as this package is cross-platform? Windows doesn't natively have a concept of `sudo` or even `root`.

File: webi-essentials/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+### Linux (Debian, Ubuntu)
```
Comment (HacDan) [inline webi-essentials/README.md:17]: Are these the manual installation instructions? If so, would it be better to heading that off similarly to other packages?

File: webi-essentials/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+```sh
+sudo apk add --no-cache \
+    curl \
+    git \
+    tar \
+    wget \
+    xz \
+    zip
+```
```
Comment (HacDan) [inline webi-essentials/README.md:40]: Are you skipping the `Files` section as this is a meta package?

**Review by HacDan — APPROVED** [review]
Only nitpicks. Nothing stopping this from going live.

File: webi-essentials/README.md (hunk 1 above)
Comment (coolaj86) [reply webi-essentials/README.md:15]: This doesn't require root on windows and it does ask if you want to use sudo before it does anything.

File: webi-essentials/README.md (hunk 2 above)
Comment (coolaj86) [reply webi-essentials/README.md:17]: What do you mean?

File: webi-essentials/README.md (hunk 3 above)
Comment (coolaj86) [reply webi-essentials/README.md:40]: It's really platform dependent, except for `xz`. Webi doesn't manage much for this.

File: webi-essentials/README.md (hunk 2 above)
Comment (coolaj86) [reply webi-essentials/README.md:17]: I understand. Will update.

**Comment by coolaj86** [comment]
I'm gonna tweak this a bit more.

I thought I had a prompt but I think I removed it due to not being sure of a reliable tty detection.

I know what to do for that now so that it prompts when the user is present, but not during a fully automated scenario.

File: webi-essentials/README.md (hunk 3 above)
Comment (coolaj86) [reply webi-essentials/README.md:98]: I added the Files section anyway. It's darn helpful as an at-a-glance reference for the macOS locations especially.

**Comment by coolaj86** [comment]
@HacDan give this another look now.

Now I'll have to go back and update `git`, `fish` and a few others with similar behavior and prompts.

Not gonna lie, I'm very satisfied to finally have the TTY detection working so that I can automate over SSH without being hindered _and_ have local prompts for politeness.

**Review by HacDan — APPROVED** [review]
LGTM. Thanks for integrating the cheat sheet changes!

**Pull request merged**